sdk/src/utils/public/apr-calculation.ts

```diff
@@ -0,0 +1,135 @@
+import Decimal from "decimal.js";
+
+export type RewardToken = {
+  emissionPerWeek: number;
+  tokenPrice: number;
+};
+
+export type PoolAPRReward = {
+  apr: number;
+  rewards: { [k: string]: number } | null;
+};
+
+/**
+ *
+ * @param emissionPerWeek : reward for weekly emission
+ * @param tokenPrice : price of token in USDT
+ * @param tvl : Liquidity is TVL
+ * @returns : Reward APR for token
+ */
+function calculateTokenPoolRewardAPR(
+  emissionPerWeek: number,
+  tokenPrice: number,
+  tvl: number
+): number {
+  return new Decimal(emissionPerWeek)
+    .mul(tokenPrice)
+    .mul(new Decimal(365).div(7))
+    .div(tvl)
+    .toNumber();
+}
+
+/**
+ *
+ * @param vol24H : volumn 24h of pool
+ * @param feeRate : fee rate in raw value Eg: 0.001% -> 0.00001
+ * @param tvl : Liquidity is TVL
+ * @param rewards : rewards tokens APR maybe contain one or more or nothing reward token
+ * @param toPercent : APR result in percentage or not
+ * @returns  see: PoolAPRReward
+ */
+export function calculatePoolAPR(
+  vol24H: number,
+  feeRate: number,
+  tvl: number,
+  rewards: { [k: string]: RewardToken } | null,
+  toPercent: boolean = true
+): PoolAPRReward {
+  // compute fee APR
+  const feeAPR = new Decimal(vol24H).mul(feeRate).mul(365).div(tvl).toNumber();
+
+  // compute reward for each token
+  if (!rewards || Object.keys(rewards).length === 0) {
+    return {
+      apr: toPercent ? new Decimal(feeAPR).mul(100).toNumber() : feeAPR,
+      rewards: null,
+    };
+  }
+  const rewardEachToken: { [k: string]: number } = {};
+  let totalReward = new Decimal(0);
+  // compute reward for each token
+  Object.entries(rewards).forEach(([key, value]) => {
+    const rewardApr = calculateTokenPoolRewardAPR(value.emissionPerWeek, value.tokenPrice, tvl);
+    totalReward = totalReward.add(rewardApr);
+    rewardEachToken[key] = toPercent ? new Decimal(rewardApr).mul(100).toNumber() : rewardApr;
+  });
+
+  const totalAPR = new Decimal(feeAPR).add(totalReward).toNumber();
+
+  return {
+    apr: toPercent ? new Decimal(totalAPR).mul(100).toNumber() : totalAPR,
+    rewards: rewardEachToken,
+  };
+}
+
+export function calculateTokenPositionRewardAPR(
+  emissionPerWeek: number,
+  tokenPrice: number,
+  share: number,
+  positionBalance: number
+): number {
+  return new Decimal(emissionPerWeek)
+    .mul(tokenPrice)
+    .mul(new Decimal(365).div(7))
+    .mul(share)
+    .div(positionBalance)
+    .toNumber();
```

Comment:
những chỗ tính toán thì em comment thêm công thức gốc để sau này đọc lại code dễ hơn nhen em. Dùng lib để tăng độ chính xác nhưng nó lại không dễ đọc á em, cho nên cần thêm phần comment công thức gốc. Em thêm comment cho những chỗ khác luôn nhen.

```suggestion
  // emissionPerWeek * tokenPrice * (365 / 7) * share / positionBalance
  return new Decimal(emissionPerWeek)
    .mul(tokenPrice)
    .mul(new Decimal(365).div(7))
    .mul(share)
    .div(positionBalance)
    .toNumber();
```